src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp

```diff
@@ -150,4 +152,124 @@ struct Mortars {
     return {Parallel::AlgorithmExecution::Continue, std::nullopt};
   }
 };
+
+/// \brief Initialize/update items related to mortars after an AMR change
+///
+/// Mutates:
+///   - Tags::MortarData<dim>
+///   - Tags::MortarMesh<dim>
+///   - Tags::MortarSize<dim>
+///   - Tags::MortarNextTemporalId<dim>
+///   - evolution::dg::Tags::NormalCovectorAndMagnitude<dim>
+///   - Tags::MortarDataHistory<dim, typename dt_variables_tag::type>>
+///
+/// For p-refinement:
+///   - Does nothing to MortarDataHistory (only valid for global time-stepping)
+///     or MortarNextTemporalId (only valid for no h-refinement)
+///   - Sets the other Mortar tags to be default initialized for each neighbor
+///   - Sets the NormalCovectorAndMagnitude to std::nullopt
+template <typename Metavariables>
+struct ProjectMortars : tt::ConformsTo<amr::protocols::Projector> {
+  static constexpr size_t dim = Metavariables::volume_dim;
+  using dt_variables_tag = typename db::add_tag_prefix<
+      ::Tags::dt, typename Metavariables::system::variables_tag>;
+  using mortar_data_history_type =
+      typename Tags::MortarDataHistory<dim,
+                                       typename dt_variables_tag::type>::type;
+
+  using Key = std::pair<Direction<dim>, ElementId<dim>>;
+
+  template <typename MappedType>
+  using MortarMap = std::unordered_map<Key, MappedType, boost::hash<Key>>;
+
+  using return_tags =
+      tmpl::list<Tags::MortarData<dim>, Tags::MortarMesh<dim>,
+                 Tags::MortarSize<dim>, Tags::MortarNextTemporalId<dim>,
+                 evolution::dg::Tags::NormalCovectorAndMagnitude<dim>,
+                 Tags::MortarDataHistory<dim, typename dt_variables_tag::type>>;
+  using argument_tags =
+      tmpl::list<domain::Tags::Mesh<dim>, domain::Tags::Element<dim>>;
+
+  static void apply(
+      const gsl::not_null<MortarMap<evolution::dg::MortarData<dim>>*>
+          mortar_data,
+      const gsl::not_null<MortarMap<Mesh<dim - 1>>*> mortar_mesh,
+      const gsl::not_null<MortarMap<std::array<Spectral::MortarSize, dim - 1>>*>
+          mortar_size,
+      const gsl::not_null<MortarMap<TimeStepId>*> /*mortar_next_temporal_id*/,
+      const gsl::not_null<
+          DirectionMap<dim, std::optional<Variables<tmpl::list<
+                                evolution::dg::Tags::MagnitudeOfNormal,
+                                evolution::dg::Tags::NormalCovector<dim>>>>>*>
+          normal_covector_and_magnitude,
+      const gsl::not_null<mortar_data_history_type*>
+      /*mortar_data_history*/,
+      const Mesh<dim>& /*new_mesh*/, const Element<dim>& new_element,
+      const std::pair<Mesh<dim>, Element<dim>>& /*old_mesh_and_element*/) {
+    static_assert(not Metavariables::local_time_stepping,
+                  "AMR with local time-stepping is not yet supported");
+
+    mortar_data->clear();
+    mortar_mesh->clear();
```

Comment:
How is this set later? My understanding is that we can set this here by keeping track of the neighbor meshes, correct?

Reply:
It is set when the mortar data is received from the neighbor.  Needs to be reactive in order for subcell to work.

Reply:
Do you have a specific change request?

Reply:
Nope, sounds good.